WordPress/Classes/Utility/ReachabilityUtils+OnlineActions.swift

```diff
@@ -35,4 +38,26 @@ extension ReachabilityUtils {
                 return notification.userInfo?[Foundation.Notification.reachabilityKey] as? Bool == true
         })
     }
+
+    /// Shows a generic non-blocking "No Connection" error message to the user.
+    ///
+    /// We use a Snackbar instead of a literal Alert because, for internet connection errors,
+    /// Alerts can be disruptive.
+    @objc static func showNoInternetConnectionNotice() {
+        let notice = Notice(title: NoConnectionMessage.title,
+                            message: NoConnectionMessage.message,
+                            tag: NoConnectionMessage.tag)
+        ActionDispatcher.dispatch(NoticeAction.post(notice))
+    }
+
+    /// Dismiss the currently shown Notice if it was created using showNoInternetConnectionNotice()
+    @objc static func dismissNoInternetConnectionNotice() {
+        if StoreContainer.shared.notice.state.notice?.tag == NoConnectionMessage.tag {
```

Comment:
This is great functionality, but I'm feeling like this should move into the notice presenter. If we changed `dismissCurrentNotice()` to be `dismissNotice(tagged:)` then we could just call the dismiss here, and let the presenter deal with the results. 

Also all the dots in `StoreContainer.shared.notice.state.notice` suggest it knows far too much about the StoreContainer 😜

Reply:
Good point. I was under the assumption that it was okay to reference `StoreContainer` anywhere since it's a singleton. I'll move it. I think we should keep this method though since the `tag` value is in this file.

Reply:
Hmm, I just remembered how I ended up with this solution. I wasn't sure of the Flux-based implementation of this. 

First, how do we show a `Notice`?

```swift
ActionDispatcher.dispatch(NoticeAction.post(notice))
```

Looks okay enough and close to Flux. Next question is how do we dismiss the current `Notice`? Based on the code above, I expected it would be like this:

```swift
ActionDispatcher.dispatch(NoticeAction.dismiss)
```

But that didn't work. The `NoticeAction.dismiss` action only prompts the `NoticeStore` to `pop` and update the current `state`. Then, the `NoticePresenter` will receive this _change_ but it only cares about the `NoticeStore.nextNotice`. With these, I believe the dismissal is inconsistent in the design because you would need the `NoticePresenter` to do it:

```swift
noticePresenter?.dismissCurrentNotice()
```

Based on my experience with Redux/Flux, it should be the UI (`NoticePresenter`) that _reacts_ to the changes in `NoticeStore` and call `dismissCurrentNotice()` if the `NoticeStore` changes. 

-end- 

Just wanted to share that. I'm still going to move this logic to the `NoticePresenter`. 😄

Reply:
I think your analysis is right. 

1. `ActionDispatcher.dispatch(NoticeAction.dismiss)` should dismiss the current notice. I think we should consider the current behaviour a bug, unless @frosty knows of a reason to keep the current functionality.

2. Given `1.`, the proper API is probably to add a `NoticeAction` to remove instances of a given tag, such as `ActionDispatcher.dispatch(NoticeAction.clear(tag: .networkError)`.

Thoughts?

Reply:
Just thinking through the dismissal... here's what currently happens:

* A new notice is at the top of the list, and a notice view is displayed for a preset amount of time
* After that time has elapsed, or if the user interacts with the notice, an animation begins to hide the notice.
* After the animation has completed, and the notice has completely disappeared from the screen, we send a dismiss action which pops the notice and checks for a new one to display.

If instead we change things around so that a dismiss action is sent on user interaction with a notice, or after the show time has expired, we'd need to do something like this:

* A new notice is at the top of the list, and a notice view is displayed for a preset amount of time
* After that time has elapsed, or if the user interacts with the notice, a `dismiss` action is dispatched.
* The store is updated to pop the current notice and populate `nextNotice` with a new one if one is available.
* We'd change `NoticePresenter`'s `store.onChange` handler to begin the animation to hide the current notice if there is one. Or, if no notice is currently displayed, display a new one (if there is one).
* If a notice was being displayed, when the animation is complete we check if there is a new notice to display, and display it.

I _think_ this should work 🤔, it's just making sure that logic at the end for doing the dismiss animation and picking up the next notice is working correctly.

I agree that firing a `dismiss` action would be a nicer way of initiating a dismiss, because then it's decoupled and we can more easily dismiss a notice from anywhere.

Reply:
Thanks, @frosty. That sounds close to what I was thinking of. 

I discussed this with @nheagy yesterday. We'll work on `NoticeAction.dismiss` on a separate PR and leave this one open for now. We'll update this once the `NoticeAction.dismiss` PR is merged.

Reply:
Sounds like a good plan!